index.ts

```diff
@@ -55,6 +57,21 @@ function useLocalStorage<T>(
     }
   }, [storedValue]);
 
+  useEffect(() => {
+    const handleStorageChange = (e: StorageEvent) => {
+      if (!syncData || e.key !== key) return;
```

Comment:
If `syncData` is false, should we just return from the effect altogether instead of adding a storage event handler?

Reply:
That seems like a good idea. The reasoning why I did it this way is completely irrelevant after thinking about this again.